src/openzeppelin/token/erc1155/presets/ERC1155MintableBurnable.cairo

```diff
@@ -0,0 +1,250 @@
+# SPDX-License-Identifier: MIT
+# OpenZeppelin Contracts for Cairo v0.2.1 (token/erc1155/presets/ERC1155MintableBurnable.cairo)
+
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.uint256 import Uint256
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.math import assert_not_zero
+
+from openzeppelin.access.ownable.library import Ownable
+from openzeppelin.token.erc1155.library import ERC1155
+from openzeppelin.introspection.erc165.library import ERC165
+
+#
+# Constructor
+#
+
+@constructor
+func constructor{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(uri: felt, owner: felt):
+    ERC1155.initializer(uri)
+    Ownable.initializer(owner)
+    return ()
+end
+
+#
+# Getters
+#
+
+@view
+func supportsInterface{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(interfaceId: felt) -> (success: felt):
+    return ERC165.supports_interface(interfaceId)
+end
+
+@view
+func uri{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(id: Uint256) -> (uri: felt):
+    return ERC1155.uri(id)
+end
+
+@view
+func balanceOf{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(account: felt, id: Uint256) -> (balance: Uint256):
+    return ERC1155.balance_of(account, id)
+end
+
+@view
+func balanceOfBatch{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(
+        accounts_len: felt,
+        accounts: felt*,
+        ids_len: felt,
+        ids: Uint256*
+    ) -> (balances_len: felt, balances: Uint256*):
+    let (balances_len, balances) =  ERC1155.balance_of_batch(accounts_len, accounts, ids_len, ids)
+    return (balances_len, balances)
+end
+
+@view
+func isApprovedForAll{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(account: felt, operator: felt) -> (isApproved: felt):
+    let (is_approved) = ERC1155.is_approved_for_all(account, operator)
+    return (is_approved)
+end
+
+@view
+func owner{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }() -> (owner: felt):
+    let (owner: felt) = Ownable.owner()
+    return (owner)
+end
+
+#
+# Externals
+#
+
+@external
```

Comment:
I just noticed that `setURI` is set to external where it is set to internal in Solidity version ([source](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/ERC1155.sol#L251)), not sure if it is a bug or a feature.

Reply:
So the MintableBurnable preset is a little misleading and maybe should be renamed, it's mostly there to expose methods for the tests, as in this case. Whether or not this should be the case is a good question, but at least the preset is somewhat sane still.